packages/react-native-media-player/android/src/main/java/com/yajananrao/mediaplayer/MediaPlayerService.java

```diff
@@ -0,0 +1,590 @@
+    }
+
+    private void setDataSource(final String uri) {
+        Log.d(TAG, "onPlayFromUri: song received " + uri);
+        try {
+            initalLoad = true;
+            if (mMediaPlayer != null && mMediaPlayer.isPlaying()) {
+                    stopPlayback();
+            }
+            initMediaPlayer();
+            mMediaPlayer.setDataSource(uri);
+            if (uri.toLowerCase().startsWith("http://") || uri.toLowerCase().startsWith("https://")) {
+                onlineResource = true;
+            } else {
+                onlineResource = false;
+            }
+            new Thread(new Runnable() {
+                public void run() {
+                    initMediaSessionMetadata(uri);
+                }
+            }).start();
+
+
+        } catch (Exception e) {
+            Log.e(TAG, "onPlayFromUri: " + e.toString());
+            return;
+        }
+    }
+
+    private BroadcastReceiver mNoisyReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (mMediaPlayer != null && mMediaPlayer.isPlaying()) {
+                mMediaPlayer.pause();
+            }
+        }
+    };
+
+    private MediaSessionCompat.Callback mMediaSessionCallback = new MediaSessionCompat.Callback() {
+
+        @Override
+        public void onPlay() {
+            Log.i(TAG, "onPlay: started");
+            super.onPlay();
+            playbackNow();
+        }
+
+        @Override
+        public void onStop() {
+            super.onStop();
+            stopPlayback();
+        }
+
+        @Override
+        public void onPause() {
+            super.onPause();
+            pausePlayback();
+        }
+
+        @Override
+        public void onPlayFromSearch(String query, Bundle extras) {
+            super.onPlayFromSearch(query, extras);
+        }
+
+        @Override
+        public void onPlayFromUri(Uri uri, Bundle extras) {
+            super.onPlayFromUri(uri, extras);
+            setDataSource(uri.toString());
+        }
+
+        @Override
+        public void onSkipToNext() {
+            Log.d(TAG, "onSkipToNext ");
+            super.onSkipToNext();
+            pausePlayback();
+            setMediaPlaybackState(PlaybackStateCompat.STATE_SKIPPING_TO_NEXT);
+        }
+
+        @Override
+        public void onSkipToPrevious() {
+            super.onSkipToPrevious();
+            pausePlayback();
+            setMediaPlaybackState(PlaybackStateCompat.STATE_SKIPPING_TO_PREVIOUS);
+        }
+
+        @Override
+        public void onSeekTo(long pos) {
+            super.onSeekTo(pos);
+            Log.d(TAG, "onSeekTo " + pos);
+            mMediaPlayer.seekTo((int) pos);
+            setMediaPlaybackState(PlaybackStateCompat.STATE_NONE);
+
+        }
+
+        @Override
+        public void onCustomAction(String action, Bundle extras) {
+            if (action.equals(ACTION_PROGRESS_UPDATE)) {
+                if (mMediaPlayer.isPlaying()) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/yajananrao/Nadha/pullRequest?prid=5728824)